packages/apps/hr/src/components/templates/ExpenseLineDisplayType/ExpenseLineDisplayType.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useCallback, useEffect, useMemo} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {useThemeColor, NumberBubble, Text} from '@axelor/aos-mobile-ui';
+import {useSelector, useTranslator, useDispatch} from '@axelor/aos-mobile-core';
+import {ExpenseLine} from '../../../types';
+import {
+  searchGeneralExpenseLines,
+  searchKilometricExpenseLines,
+} from '../../../features/expenseLineSlice';
+import ExpenseLineTypeSwitch from '../ExpenseLineTypeSwitch/ExpenseLineTypeSwitch';
+
+interface ExpenseLineDisplayTypeProps {
+  isAddButton?: boolean;
+  onChange: (mode: any) => void;
+}
+
+const ExpenseLineDisplayType = ({
+  isAddButton,
+  onChange,
+}: ExpenseLineDisplayTypeProps) => {
+  const dispatch = useDispatch();
+  const I18n = useTranslator();
+  const Colors = useThemeColor();
+
+  const {expense} = useSelector((state: any) => state.expense);
+  const {
+    generalExpenseLineList,
+    totalNumberExpenseGeneral,
+    kilometricExpenseLineList,
+    totalNumberExpenseKilomectric,
+  } = useSelector((state: any) => state.expenseLine);
+
+  useEffect(() => {
+    dispatch(
+      (searchKilometricExpenseLines as any)({expenseId: expense?.id, page: 0}),
+    );
+
+    dispatch(
+      (searchGeneralExpenseLines as any)({expenseId: expense?.id, page: 0}),
+    );
+  }, [dispatch, expense]);
+
+  const displayToggle = useMemo(() => {
+    return totalNumberExpenseGeneral > 0 && totalNumberExpenseKilomectric > 0;
+  }, [totalNumberExpenseGeneral, totalNumberExpenseKilomectric]);
+
+  const isGeneral = useMemo(
+    () =>
+      generalExpenseLineList != null &&
+      totalNumberExpenseGeneral > 0 &&
+      totalNumberExpenseKilomectric === 0,
+    [
+      generalExpenseLineList,
+      totalNumberExpenseGeneral,
+      totalNumberExpenseKilomectric,
+    ],
+  );
+
+  const isKilometric = useMemo(
+    () =>
+      kilometricExpenseLineList != null &&
+      totalNumberExpenseKilomectric > 0 &&
+      totalNumberExpenseGeneral === 0,
+    [
+      kilometricExpenseLineList,
+      totalNumberExpenseGeneral,
+      totalNumberExpenseKilomectric,
+    ],
+  );
+
+  const hasExpenseLines = useMemo(() => {
+    return !(!isGeneral && !isKilometric);
+  }, [isGeneral, isKilometric]);
```

Comment:
You can't use `isGeneral` and `isKilometric` for this boolean, because if there are expense lines of general and kilometric type the function will return false even though there are